client-app/broadcast.ts

```diff
@@ -32,4 +37,14 @@ export function setupBroadcastGlobalListeners() {
       await fetchShortCart();
     }
   });
+  on(unauthorizedErrorEvent, async () => {
+    await fetchUser();
+    if (!isPasswordNeedToBeChanged.value) {
+      const { hash, pathname, search } = location;
+
+      if (pathname !== "/sign-in") {
+        location.href = `/sign-in?returnUrl=${pathname + search + hash}`;
```

Comment:
Shouldn't we use `router` here and below where you use `location`?

Reply:
The App should be reloaded to clear state